datafusion/substrait/src/serializer.rs

```diff
@@ -26,9 +26,30 @@ use substrait::proto::Plan;
 
 use std::fs::OpenOptions;
 use std::io::{Read, Write};
+use std::path::Path;
 
-pub async fn serialize(sql: &str, ctx: &SessionContext, path: &str) -> Result<()> {
+/// Plans a sql and serializes the generated logical plan to bytes.
+/// The bytes are then written into a file at `path`.
+///
+/// Returns an error if the file already exists and is not empty.
+pub async fn serialize(
+    sql: &str,
+    ctx: &SessionContext,
+    path: impl AsRef<Path>,
+) -> Result<()> {
     let protobuf_out = serialize_bytes(sql, ctx).await;
+
+    match std::fs::metadata(path.as_ref()) {
+        Ok(meta) if meta.len() > 0 => {
+            return Err(DataFusionError::Substrait(format!(
+                "Failed to encode substrait plan: the file {} already exists and is not empty", 
```

Comment:
You could remove `substrait` here because it's already part of the error prefix https://github.com/apache/datafusion/blob/19fe44cf2f30cbdd63d4a4f52c74055163c6cc38/datafusion/common/src/error.rs#L517
```suggestion
                "Failed to encode plan: the file {} already exists and is not empty", 
```

Reply:
Adding a `substrait` is because there already had an error message "Failed to encode substrait plan" in the fuction `serialize_bytes`. I was following that error message to make error messages consistent.

Reply:
Removed the word "substrait" from all error messages.